cub/test/catch2_test_device_reduce.cu

```diff
@@ -46,6 +46,21 @@ DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMin, device_arg_min);
 DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Max, device_max);
 DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMax, device_arg_max);
 
+#if defined(__CUDACC_VER_MAJOR__) && defined(__CUDACC_VER_MINOR__) && ((__CUDACC_VER_MAJOR__ > 11) || ((__CUDACC_VER_MAJOR__ == 11) && (__CUDACC_VER_MINOR__ >= 5)))
+#define HAS_NV_PREFIX
+#endif
```

Comment:
Q: How is `HAS_NV_PREFIX` different than `__NVCC_DIAG_PRAGMA_SUPPORT__`? Why do we need this separate workaround?